test-data/unit/check-newsemanal.test

```diff
@@ -995,6 +995,32 @@ class C(type):
 
 reveal_type(A.f()) # E: Revealed type is 'builtins.int'
 
+[case testNewAnalyzerMetaclassSix1]
+import six
+
+class A(six.with_metaclass(B)):
+    pass
+
+class B(type):
+    def f(cls) -> int:
+        return 0
+
+reveal_type(A.f()) # E: Revealed type is 'builtins.int'
+
+[case testNewAnalyzerMetaclassSix2]
+import six
+
+@six.add_metaclass(B)
+class A:
+    pass
+
+class B(type):
+    def f(cls) -> int:
+        return 0
+
+reveal_type(A.f()) # E: Revealed type is 'builtins.int'
```

Comment:
I would add couple more tests with more tricky deferrals. For example with a base class that also gets deferred.

Also since there are now only 6 tests fail in `check-classes.test`, maybe just explicitly skip those with `# flags: --no-new-semantic-analyzer` and a short motivational comment or a link to issue?